src/test/java/tutorspet/logic/commands/DeleteAttendanceCommandTest.java

```diff
@@ -70,30 +61,18 @@ public void constructor_nullWeek_throwsNullPointerException() {
     }
 
     @Test
-    public void execute_validIndexes_success() {
+    public void execute_validIndexes_success() throws CommandException {
         Index moduleClassIndex = INDEX_FIRST_ITEM;
         Index lessonIndex = INDEX_FIRST_ITEM;
         Index studentIndex = INDEX_FIRST_ITEM;
         Week targetWeek = VALID_WEEK_1;
 
         ModuleClass moduleClass = model.getFilteredModuleClassList().get(moduleClassIndex.getZeroBased());
         Student student = model.getFilteredStudentList().get(studentIndex.getZeroBased());
-        Lesson lesson = moduleClass.getLessons().get(lessonIndex.getZeroBased());
 
-        // delete week 1 attendance record
-        Map<UUID, Attendance> record = lesson.getAttendanceRecordList()
-                .getAttendanceRecord(targetWeek).getAttendanceRecord();
-        Map<UUID, Attendance> updatedRecord = new HashMap<>(record);
-        updatedRecord.remove(student.getUuid());
-        AttendanceRecord updatedAttendanceRecord = new AttendanceRecord(updatedRecord);
-        List<AttendanceRecord> updatedAttendanceRecords =
-                new ArrayList<>(lesson.getAttendanceRecordList().getAttendanceRecordList());
-        updatedAttendanceRecords.set(targetWeek.getZeroBasedWeekIndex(), updatedAttendanceRecord);
-
-        Lesson modifiedLesson = new LessonBuilder(lesson)
-                .withAttendanceRecordList(new AttendanceRecordList(updatedAttendanceRecords)).build();
         ModuleClass modifiedModuleClass =
-                manualReplaceLessonToModuleClass(moduleClass, lesson, modifiedLesson);
+                deleteAttendanceFromModuleClass(moduleClass, lessonIndex, VALID_WEEK_1, student);
```

Comment:
Perhaps can change to `targetWeek`
```suggestion
                deleteAttendanceFromModuleClass(moduleClass, lessonIndex, targetWeek, student);
```